assemblyline_v4_service/common/keytool_parse.py

```diff
@@ -0,0 +1,62 @@
+import re
+from subprocess import Popen, PIPE, call
+from assemblyline.common.str_utils import safe_str
+
+class Certificate():
+    def __init__(self):
+        self.raw = ""
+        self.issuer = ""
+        self.owner = ""
+        self.country = ""
+        self.valid_from = ""
+        self.valid_to = ""
+
+def keytool_printcert(cert_path: str):
+    """
+    This function runs the 'keytool -printcert' command against a provided file
+
+    :param cert_path: A path to a certificate
+    :return: the string output of 'keytool -printcert' or None
+    """
+    stdout, stderr = Popen(["keytool", "-printcert", "-file", cert_path],
+                                stderr=PIPE, stdout=PIPE).communicate()
+    stdout = safe_str(stdout)
+
+    if stdout:
+        if "keytool error" not in stdout:
+            return stdout
+
+    return None
+
+def certificate_chain_from_printcert(printcert: str):
+    """
+    This function parses the output of 'keytool -printcert' and creates a list of Certificate objects.
+    The input to this function is the output of keytool_printcert
+
+    :param printcert: the string output of 'keytool -printcert'
+    :return: a list of the parsed out certificates. If only one certificate is present and not a chain, then the list will have one element.
+    """
+    certs = []
+
+    for cert_str in  re.split('Certificate\[\d+\]:', printcert): # split printcert output incase of certificate chain
+        if cert_str == '':
+            continue
+        cert = Certificate()
+        cert.raw = cert_str.strip()
+        for line in cert_str.splitlines():
+            if "Owner:" in line:

```

Comment:
is it possible for a line to have `Owner:`, `Issuer:` and `Valid from:` in it? If not then these consecutive `if` statements should be converted to `if, elif, elif`